examples/with-styled-components/pages/_document.js

```diff
@@ -1,21 +1,28 @@
 import Document, { Head, Main, NextScript } from 'next/document'
-import { ServerStyleSheet } from 'styled-components'
 
 export default class MyDocument extends Document {
+  static getInitialProps ({ renderPage, req }) {
+    const isServer = !!req
+    if (isServer) {
+      // eslint-disable-next-line no-eval
+      const ServerStyleSheet = eval("require('styled-components').ServerStyleSheet")
```

Comment:
I'm doing this b/c of @arunoda's arunoda.me/blog/ssr-and-server-only-modules, I hope this is correct. :blush:

Reply:
The question is, is it fine to not add the `styleTags` when we're inside the browser? This seems like it to me as we end up with two sets of our styles on the styled-components website... So maybe we can just return empty styles in the browser by escaping, and only run this on the server:

```js
// Not sure if we should rely on req in getInitialProps
const isServer = typeof process !== 'undefined' && !process.browser;

export default class MyDocument extends Document {
  static getInitialProps ({ renderPage, req }) {
    if (!isServer) {
      return { page: renderPage(), styleTags: null }
    }

    const sheet = new ServerStyleSheet()
    // Bind is not enough as we receive a component and not an element
    // https://github.com/styled-components/styled-components-website/blob/master/pages/_document.js#L151
    const page = renderPage(App => props => sheet.collectStyles(<App {...props} />))
    const styleTags = sheet.getStyleTags()
    return { ...page, styleTags }
}
```

Also, I'd leave that `eval` stuff out. `ServerStyleSheet` is present (or tree shaken; or uglified away) in the client bundle anyway, and we're not dealing with a server bundle. So this will only incur an "eval-cost" without benefits